src/ORM/DataObject.php

```diff
@@ -3422,7 +3422,13 @@ public static function get_one($callerClass = null, $filter = "", $cache = true,
 
         $item = null;
         if (!$cache || !isset(self::$_cache_get_one[$callerClass][$cacheKey])) {
-            $dl = DataObject::get($callerClass)->where($filter)->sort($orderBy);
+            $dl = DataObject::get($callerClass);
+            if (!empty($filter)) {
+                $dl = $dl->where($filter);
+            }
+            if (!empty($orderBy)) {
```

Comment:
```suggestion
            if (!empty($orderBy) || $sort === null) {
```

Reply:
There is no $sort variable.  If you meant `$orderBy === null` instead, then there's no reason to call sort(null) as it's a fresh datalist that won't have a prior sort.  We're probably best just adding strong typing to this method and disallow null alltogether

Reply:
Sorry, yes, I meant `$orderBy`

> it's a fresh datalist that won't have a prior sort

`DataObject`s can declare a _default_ sort. Calling `sort(null) ` would clear that default sort.

Reply:
Updated